### PR TITLE
Improve robustness of checking tags of allocated memory blocks

### PR DESCRIPTION
**Pull request by keithc-ca**
J9MemTag.allocSize has type uintptr_t (aka UDATA), which is mapped to U64 by MSVC on Windows (or with the changes in eclipse/omr#3355 on all 64-bit platforms). A random address treated as a pointer to a J9MemTag has a 50% chance of referring to a J9MemTag where allocSize >= 2^63 which would trigger an InvalidDataTypeException from U64.longValue(). This change modifies ROUNDED_FOOTER_OFFSET() so it always returns a UDATA object independent of the actual
argument type, thus avoiding that exception for blocks that otherwise would fail the sanity checks.

https://github.com/eclipse/omr/pull/3422 replays eclipse/omr#3355.

### CODE REVIEW

**Comment by keithc-ca** [comment]
Jenkins test all zlinux jdk11 depends eclipse/omr#3422

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java [hunk 1]
```diff
@@ -266,8 +265,9 @@ public static VoidPointer j9mem_get_footer_padding(J9MemTagPointer headerEyeCatc
 	
 	public static UDATA ROUNDED_FOOTER_OFFSET(UDATA number) 
 	{
-		UDATA a = (number.add(ROUNDING_GRANULARITY - 1)).add(J9MemTag.SIZEOF);
-		UDATA b = new UDATA(ROUNDING_GRANULARITY - 1).bitNot();
+		UDATA mask = new UDATA(ROUNDING_GRANULARITY - 1);
+		UDATA a = mask.add(number).add(J9MemTag.SIZEOF);
+		UDATA b = mask.bitNot();
```
Comment (DanHeidinga) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java]: Based on the PR description
> This change modifies ROUNDED_FOOTER_OFFSET() so it always returns a UDATA object independent of the actual
argument type, thus avoiding that exception for blocks that otherwise would fail the sanity checks.

I'm not clear on why the original code wouldn't have already always returned a UDATA.  Can you explain why this change is required?

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java]: U64.add() returns a U64 while UDATA.add(UScalar) returns UDATA; UDATA.longValue() doesn't throw an exception.

Comment (DanHeidinga) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java]: Which variable is a U64 here? `J9MemTag.SIZEOF`?

**Comment by pshipton** [comment]
Jenkins test all win jdk11 depends eclipse/omr#3422

**Comment by pshipton** [comment]
jenkins test sanity win jdk11 depends eclipse/omr#3422

**Comment by pshipton** [comment]
jenkins test extended win jdk11 depends eclipse/omr#3422

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java (hunk 1 above)
Comment (DanHeidinga) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java]: Thanks Keith.  I had forgotten that `U64` was changed to be a subclass of `UDATA`
